package.json

```diff
@@ -29,10 +29,10 @@
   "author": "Kent C. Dodds <kent@doddsfamily.us> (http://kentcdodds.com/)",
   "license": "MIT",
   "dependencies": {
-    "babel-core": "^6.26.3",
-    "babel-plugin-macros": "^2.2.1",
-    "babel-register": "^6.26.0",
-    "babel-template": "^6.26.0",
+    "@babel/core": "^7.0.0-beta.49",
+    "@babel/register": "^7.0.0-beta.49",
+    "@babel/template": "^7.0.0-beta.49",
```

Comment:
I'm concerned about hard-coding these values. Is there a way that we can have a more generous range that will ensure that people using this package with their own babel versions installed will install the same version that this package installs so we don't have issues with multiple versions being installed in their project?

Reply:
That is a very good question. I’ve been thinking about this too.
I think the upgrade story for a lib like this is a bit bumpy.

The problem with version ranges is the moving around of packages.
babel-core -> @babel/core etc.

Stuff that is only used internally like @babel/template shouldn’t actually matter I guess, because the behaviour is completely encapsulated (?).

It gets more interesting with @babel/core - I’m not super aware of how this works, but I guess that v7 does not work with v6 plugins defined in .babelrc? Maybe?

If that’s true, we need to get creative with peerDependecies (optionalDependencies even, cause of the name change?) here.

Any ideas?

Reply:
I wonder if we could actually refactor the code to use only the `babel` object that we get [here](https://github.com/djfarly/babel-plugin-codegen/blob/b41bb53428b8687de5b279e79349222cb8c68ea5/src/index.js#L6). That object is whatever `babel-core` or `@babel/register` would be, regardless of what version people have. It also has a `template` function on it so we could drop `@babel/core` and `@babel/template`.

As for `@babel/register`, I'm not certain how to do that. Maybe we could do a try/catch around a require for `@babel/register` and if that fails then try to require `babel-register`, and if that fails then don't worry about it? Then we just document that if you want your codegen code to be transpiled first, you need to install one of those packages. I think that's reasonable.

Reply:
If you want to use the one they have you'd use a peerDep although not sure how you solve the peerDep issue for 2 differently named packages 😄(we did babel-core bridge for jest and others though)

Reply:
Yeah, let's just drop the babel-register thing entirely.